### ISSUE TITLE
ModuleNotFoundError: No module named '_cnest'

### ISSUE THREAD

**Issue by gyuta** [HorizonRobotics/alf#1427]
I canot find cnest github repository, so I submit here.

my environment
- mac os 12.6.2 (intel chip)
- python 3.7.16
- cnest 1.0.4

I got `ModuleNotFoundError: No module named '_cnest'` error when `import cnest`.

so, I checked cnest package dir and found no `_cnest.hogehoge.so` file.

so, I built by myself and copied it to my working dir.
```
cd <cnest dir>
c++ -O3 -Wall -shared -std=c++11 -undefined dynamic_lookup $(python3 -m pybind11 --includes) nest.cpp -o _cnest$(python3-config --extension-suffix)
cp _cnest.hogehoge.so <working dir>
```
and I can excute `import cnest` without error.

But I think this procedure is too tedious.

So does anyone know more simple solution?

**Comment by hnyu** [comment]
Hi @gyuta , can you try directly installing the repo: `pip install git+https://github.com/HorizonRobotics/cnest.git`?

**Comment by gyuta** [comment]
@hnyu 

Thanks for quick reply!!

I tried it but doesnt work

```
$ pip install git+https://github.com/HorizonRobotics/cnest.git                                        
Collecting git+https://github.com/HorizonRobotics/cnest.git
  Cloning https://github.com/HorizonRobotics/cnest.git to /private/var/folders/8_/_xtn7y3s07gb38gjlhgk3l4r0000gp/T/pip-req-build-rmaslzvl
  Running command git clone --filter=blob:none --quiet https://github.com/HorizonRobotics/cnest.git /private/var/folders/8_/_xtn7y3s07gb38gjlhgk3l4r0000gp/T/pip-req-build-rmaslzvl
  Resolved https://github.com/HorizonRobotics/cnest.git to commit 630d21239e282765896b6a040593066d3b796265
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: pybind11<3.0.0,>=2.6.2 in /Users/myname/.local/share/virtualenvs/mac-exp-iOgqb1lD/lib/python3.7/site-packages (from cnest==1.0.4) (2.10.3)
Building wheels for collected packages: cnest
  Building wheel for cnest (pyproject.toml) ... done
  Created wheel for cnest: filename=cnest-1.0.4-cp37-cp37m-macosx_12_0_x86_64.whl size=6800 sha256=6bcf6e5645cfe2236f4dbd5ba00c8d8ba5e46083e1caebd8839314aa81aa783d
  Stored in directory: /private/var/folders/8_/_xtn7y3s07gb38gjlhgk3l4r0000gp/T/pip-ephem-wheel-cache-x21cxmpa/wheels/8e/1c/11/5d4672f85fc2f670d0db9666a56d8c7c521a230b0e327993bc
Successfully built cnest
Installing collected packages: cnest
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
alf 0.0.6 requires cnest==1.0.5, but you have cnest 1.0.4 which is incompatible.
Successfully installed cnest-1.0.4
```

```
$ ls  /Users/myname/.local/share/virtualenvs/mac-exp-iOgqb1lD/lib/python3.7/site-packages/cnest 
__init__.py     __pycache__     format.py       nest.cpp
```

**Comment by hnyu** [comment]
@gyuta  Can you first uninstall cnest completely and try installing from the github repo link again? 

One thing to note that the PyPi cnest package will try installing cnest wheels which were only tested for certain Ubuntu versions. I noticed that you're using a MacOS system, so that wheel might not be compatible. One solution is to directly installing from the github (build from source).

（A backup solution is to git clone the cnest repo, and try `pip install -e .` inside cnest dir and see if the .so file exists. But certainly this is not the most convenient way of installing cnest.)

**Comment by hnyu** [comment]
> @hnyu
> 
> Thanks for quick reply!!
> 
> I tried it but doesnt work
> 
> ```
> $ pip install git+https://github.com/HorizonRobotics/cnest.git                                        
> Collecting git+https://github.com/HorizonRobotics/cnest.git
>   Cloning https://github.com/HorizonRobotics/cnest.git to /private/var/folders/8_/_xtn7y3s07gb38gjlhgk3l4r0000gp/T/pip-req-build-rmaslzvl
>   Running command git clone --filter=blob:none --quiet https://github.com/HorizonRobotics/cnest.git /private/var/folders/8_/_xtn7y3s07gb38gjlhgk3l4r0000gp/T/pip-req-build-rmaslzvl
>   Resolved https://github.com/HorizonRobotics/cnest.git to commit 630d21239e282765896b6a040593066d3b796265
>   Installing build dependencies ... done
>   Getting requirements to build wheel ... done
>   Preparing metadata (pyproject.toml) ... done
> Requirement already satisfied: pybind11<3.0.0,>=2.6.2 in /Users/myname/.local/share/virtualenvs/mac-exp-iOgqb1lD/lib/python3.7/site-packages (from cnest==1.0.4) (2.10.3)
> Building wheels for collected packages: cnest
>   Building wheel for cnest (pyproject.toml) ... done
>   Created wheel for cnest: filename=cnest-1.0.4-cp37-cp37m-macosx_12_0_x86_64.whl size=6800 sha256=6bcf6e5645cfe2236f4dbd5ba00c8d8ba5e46083e1caebd8839314aa81aa783d
>   Stored in directory: /private/var/folders/8_/_xtn7y3s07gb38gjlhgk3l4r0000gp/T/pip-ephem-wheel-cache-x21cxmpa/wheels/8e/1c/11/5d4672f85fc2f670d0db9666a56d8c7c521a230b0e327993bc
> Successfully built cnest
> Installing collected packages: cnest
> ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
> alf 0.0.6 requires cnest==1.0.5, but you have cnest 1.0.4 which is incompatible.
> Successfully installed cnest-1.0.4
> ```
> 
> ```
> $ ls  /Users/myname/.local/share/virtualenvs/mac-exp-iOgqb1lD/lib/python3.7/site-packages/cnest 
> __init__.py     __pycache__     format.py       nest.cpp
> ```

Hi @gyuta, we've published a new cnest version 1.0.8 and fixed an issue of building the so file PR #1434 . Please give it a try and see if it also solves your problem. Thanks!

**Comment by gyuta** [comment]
I apologize for the delay in my response. Thank you again for addressing the issue. I have confirmed that it is now working properly. Thank you very much.
